webpack/webpack.common.js

```diff
@@ -0,0 +1,54 @@
+const path = require("path");
+const srcScripts = '../src/scripts/';
+const CopyPlugin = require('copy-webpack-plugin');
+const ForkTsCheckerWebpackPlugin = require('fork-ts-checker-webpack-plugin');
+
+const outpath = path.resolve(__dirname, '../dist/');
+
+module.exports = {
+    entry: {
+        background: path.join(__dirname, srcScripts + "/background.js"),
+        content: path.join(__dirname, srcScripts + "/content.js"),
+        main: path.join(__dirname, srcScripts + "/main.js"),
+        options: path.join(__dirname, srcScripts + "/options.js"),
+        popup: path.join(__dirname, srcScripts + "/popup.js"),
+    },
+    output: {
+        path: outpath,
+        filename: 'scripts/[name].js',
+    },
+    module: {
+        rules: [
+            {
+                test: /\.tsx?$/,
+                loader: 'ts-loader',
+                exclude: /node_modules/,
+                options: {
+                    transpileOnly: true,
+                },
+            },
+        ],
+    },
+    // Setup @src path resolution for TypeScript files
+    resolve: {
+        extensions: [".ts", ".tsx", ".js"],
+        alias: {
+            "@src": path.resolve(__dirname, "../src/"),
+        },
+    },
+    plugins: [
+        new ForkTsCheckerWebpackPlugin(),
+        new CopyPlugin({
+            patterns: [
+                {
+                    from: './',
+                    to: outpath,
+                    globOptions: {
+                        ignore: ['**/scripts'],
+                    },
+                    context: 'src/',
```

Comment:
It is probably more useful to move asset files out of `src/` entirely, since there is no requirement for `src/` to be mountable as an extension any longer.

Then typescript won't have to even skip html / css files, and this plugin won't have to skip the script directory.

Reply:
I actually had this on a previous version of my branch. I had moved all of the assets to `public/` and just the scripts were in `src/`. I decided against it in case the typescript tooling didn't want to be incorporated, you could still load the extension from src without using `npm`.

Reply:
I think it makes sense to only put things in src that get compiled. I don't think it is a major concern that the dev tooling is now required to do development.

Reply:
I will leave this one to last. If it's decided the TS tooling is a go and npm is required to dev, then I can move assets to a separate file.